### ISSUE TITLE
Missing wind forcing/bottom drag functional primitives: double_gyre_wind_curl, bottom_drag

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#148]
## Background

Somax exposes public functional primitives for wind forcing and bottom drag:

```python
# somax/_src/models/qg/forcing.py

def calculate_wind_forcing(domain, params, H_0, tau0=0.08/1000.):
    """
    F_wind = (tau0/H0) * 2*pi/Ly * sin(2*pi*(y-y0)/Ly) / double-gyre wind curl pattern
    """
    Ly = domain.Lx[-1]
    y_coords_center = center_avg_2D(domain.grid_axis[-1])
    curl_tau = tau0 * 2 * math.pi / Ly * jnp.sin(2 * math.pi * (y_coords_center-params.y0) / Ly)
    return curl_tau / H_0


def calculate_bottom_drag(psi, domain, H_z=1.0, delta_ek=2.0, f0=9.375e-5, masks_psi=None):
    """
    F_drag = (delta_ek * f0^2 / 2Hz) * laplacian(psi)
    """
    omega = laplacian_batch(psi, domain.dx)
    bottom_drag_coeff = delta_ek / H_z * f0 / 2.0
    return -bottom_drag_coeff * omega[-1]
```
Source: [`somax/_src/models/qg/forcing.py`](https://github.com/jejjohnson/somax/blob/8dc7229fa57dde0366986b97b6fe7372e3da21e0/somax/_src/models/qg/forcing.py)

finitevolX currently only implements these wind forcing/drag patterns in scripts. There are **no public functional exports** for wind curl or bottom drag.

## Request

Add public functional primitives to finitevolX for:
- `double_gyre_wind_curl(y, tau0, Ly, y0, H_0)` — wind curl forcing (double-gyre pattern) at any grid location
- `bottom_drag(psi, delta_ek, H_z, f0)` — bottom drag at any grid location
      - Must support batch mode for multi-layer psi
      - Must support/clarify masks (land/sea masking) and boundary handling

**Open questions for future clarification:**
- Should Ly/y0/H_0 be derived from grid/field object or passed explicitly?
- Should output shape match finitevolX's interior/ghost ring conventions?
- Should wind curl primitive provide both centered and staggered forms?
- Any need for extension to general wind patterns (beyond double-gyre)?

## References
- Somax wind and drag implementation: [`forcing.py`](https://github.com/jejjohnson/somax/blob/8dc7229fa57dde0366986b97b6fe7372e3da21e0/somax/_src/models/qg/forcing.py)
- finitevolX scripts: search for wind curl pattern and bottom drag in QG/SWE examples/benchmarks

**Goal:**
Enable migration of Somax wind/drag functional pattern calls to finitevolX primitives for model and script use.

---

**Related labels:** enhancement
